### PR TITLE
fix: add content build run --force flag

### LINKED ISSUE #603: Add a hidden `--force` flag for `rsconnect content build rm`
It is possible for a locally tracked content build to become out-of-sync with the state of the content items on the connect server. When this happens, we should provide an escape hatch for customers to cleanup their local state file without having to modify the json file directly.

https://github.com/posit-dev/rsconnect-python/commit/f33f14ba9b65a9e1bcea7ba179ee863738312e15

### PR DESCRIPTION
**Pull request by lucasrod16**
## Intent

Provide a way to recover from `already a build running` errors without having to manually edit the JSON state file

Fixes #603 #620 https://github.com/rstudio/connect/issues/29298

## Type of Change

- [x] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach

- add a `--force` flag to the `content build run` command. this ignores `rsconnect_build_running` being set to `true` in the state file to allow users to recover from `already a build running` errors without having to manually edit the state file

- remove the `rsconnect_build_running` check from the `content build [add,remove]` commands

- `--force` will only build `NEEDS_ BUILD` content by default, but can be combined with `--retry` to build content in other states

- update deprecated `logger.warn` usage to `logger.warning`
   - https://docs.python.org/3/library/logging.html#logging.Logger.warning
   > Note
      There is an obsolete method warn which is functionally identical to warning. As warn is deprecated, please do not use it use warning instead.

## Automated Tests

added tests for `--force` and `already a build running` conditions

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [ ] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: rsconnect/actions_content.py [hunk 1]
```diff
@@ -154,7 +157,8 @@ def build_start(
         build_add_content(connect_server, all_content)
     else:
         # --retry is shorthand for --aborted --error --running
-        if retry:
+        # --force has the same behavior as --retry and also ignores when rsconnect_build_running=true
```
[the remark this answers is not in the document]
Comment (aronatkins) [reply rsconnect/actions_content.py:160]: Replying to my own comment: Any items that have `NEEDS_BUILD` are automatically included. Based on that, how do you feel about _not_ making `--force` act like `--retry`? Yes, folks can combine `--force` and `--retry`, but by default, `--force` runs only the items that have not yet been run.

**Review by aronatkins — APPROVED** [review]
Some suggestions; take if you feel they're worthwhile.

File: rsconnect/actions_content.py [hunk 2]
```diff
@@ -141,10 +131,23 @@ def build_start(
...
     poll_wait: int = 1,
     debug: bool = False,
+    force: bool = False,
 ):
     build_store = ensure_content_build_store(connect_server)
-    if build_store.get_build_running():
-        raise RSConnectException("There is already a build running on this server: %s" % connect_server.url)
+    if build_store.get_build_running() and not force:
+        raise RSConnectException(
+            "There is already a build running on this server: %s. "
```
[the remark this answers is not in the document]
Comment (lucasrod16) [reply rsconnect/actions_content.py]: I like it! Your suggested wording is more clear and accurate. Updated to use suggested wording

File: rsconnect/actions_content.py [hunk 3]
```diff
@@ -141,10 +131,23 @@ def build_start(
...
+            "Use the '--force' flag to override this check." % connect_server.url
+        )
+
+    # prompt the user to confirm that they want to --force a build.
+    if force:
+        logger.warning("Please ensure a build is not already running in another terminal before proceeding.")
+        user_input = input("Proceed with the build? Type 'yes' to confirm, any other key to cancel: ").strip().lower()
```
[the remark this answers is not in the document]
Comment (lucasrod16) [reply rsconnect/actions_content.py]: Agreed. It does feel odd to prompt after providing a `--force` flag

[the remark this answers is not in the document]
Comment (lucasrod16) [reply rsconnect/actions_content.py]: Updated to remove the prompt

File: rsconnect/actions_content.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucasrod16) [reply rsconnect/actions_content.py]: Yep, `NEEDS_BUILD` content is automatically included in the build. My thinking was that folks who end up in situations where the `--force` flag is needed likely are in a similar state that also requires using `--retry`, but I have no evidence to back that up and is yet to be seen. I think `--force` only building `NEEDS_BUILD` content is a sensible default. Updated `--force` to only build `NEEDS_BUILD` content

File: README.md [hunk 4]
```diff
@@ -885,6 +885,8 @@ build all "tracked" content that has the status `NEEDS_BUILD`.
 
 > To re-run failed builds, use `rsconnect content build run --retry`. This will build
 all tracked content in any of the following states: `[NEEDS_BUILD, ABORTED, ERROR, RUNNING]`.
+>
+> If you encounter an error stating that a build operation is already running, you can use `rsconnect content build run --force`. This will override this check and build any content marked as `NEEDS_BUILD`.
```
Comment (aronatkins) [inline README.md:889]: Should this include something like, "make sure any previously launched `content build` operation is no longer active"?

**Review by aronatkins — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (lucasrod16) [reply README.md]: Ahh yeah that's a good suggestion. Added a note for that and made an attempt to improve the flow of the wording as well

**Review by aronatkins — APPROVED** [review]

**Pull request merged**